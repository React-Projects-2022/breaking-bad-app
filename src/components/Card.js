import PropTypes from "prop-types";
import { Button } from ".";
import { openExternalURL } from "../helpers/navigate";

const Card = ({ character }) => {
  const goToMoreDetails = (url) => openExternalURL(url);
  return (
    <div className="col" key={character.char_id}>
      <div className="card">
        <div className="image-section">
          <img
            src={character.img}
            className="card-img-top"
            alt={character.name}
          />
          <span className="bottom-left">{character.portrayed}</span>
        </div>
        <div className="card-body">
          <h5 className="card-title">{character.name}</h5>
          {
            (character.char_id === undefined) ?
            (
              <>
              <p className="card-text truncate">{character.description}</p>
              
              <Button
                key={`Wiki_${character.id}`}
                label={"Más información"}
                param={character.url}
                action={goToMoreDetails}
              />
              </>
            ): (
              <>
              <p className="card-text">{character.occupation.join(", ")}</p>
              <p className="card-text">
                <small className="text-muted">
                  Appearance Seasons: {character.appearance.join(", ")}
                </small>
              </p>
              </>
            )
          }
          
        </div>
      </div>
    </div>
  )
};

// Para obligar pasar el argumento para el objeto
Card.propTypes = {
  character: PropTypes.object.isRequired,
};
export default Card;
