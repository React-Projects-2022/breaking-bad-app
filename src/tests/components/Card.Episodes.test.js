import Card from "../../components/Card";
import { shallow } from "enzyme";

describe("Pruebas de <Card /> en 'All Characters'", () => {
  const character = {
    char_id: 3,
    name: "Skyler White",
    birthday: "08-11-1970",
    occupation: [
      "House wife",
      "Book Keeper",
      "Car Wash Manager",
      "Taxi Dispatcher",
    ],
    img: "https://s-i.huffpost.com/gen/1317262/images/o-ANNA-GUNN-facebook.jpg",
    status: "Alive",
    nickname: "Sky",
    appearance: [1, 2, 3, 4, 5],
    portrayed: "Anna Gunn",
    category: "Breaking Bad",
    better_call_saul_appearance: [],
  };
  const wrapper = shallow(
    <Card character={character} useLabel={"Appearance Seasons:"} />
  );

  test("Debe de mostrar <Button /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar que las propiedades de imagen correctas", () => {
    const imageProps = wrapper.find("ImageRenderer").props();
    expect(imageProps.url).toBe(character.img);
    expect(imageProps.title).toBe(character.name);
    expect(imageProps.customClass).toBe("card-img-top");
  });

  test("Comprobar que texto del nombre de personaje es correcto", () => {
    expect(wrapper.find("h5").text()).toBe(character.name);
  });

  test("Comprobar que las ocupaciones se renderizan bien en el formato deseado", () => {
    expect(wrapper.find("p").at(0).text()).toBe(
      character.occupation.join(", ")
    );
  });

  test("Comprobar que las temporadas en que aparece se renderizan bien en el formato deseado", () => {
    expect(wrapper.find("p").at(1).text()).toBe(
      `Appearance Seasons: ${character.appearance.join(", ")}`
    );
  });
});
