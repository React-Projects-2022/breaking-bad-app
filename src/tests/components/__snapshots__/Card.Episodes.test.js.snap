// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <Card /> en 'All Characters' Debe de mostrar <Button /> correctamente 1`] = `
<div
  className="col"
  key="3"
>
  <div
    className="card"
  >
    <div
      className="image-section"
    >
      <ImageRenderer
        customClass="card-img-top"
        height={470}
        thumb="https://s-i.huffpost.com/gen/1317262/images/o-ANNA-GUNN-facebook.jpg"
        title="Skyler White"
        url="https://s-i.huffpost.com/gen/1317262/images/o-ANNA-GUNN-facebook.jpg"
        width={600}
      />
      <span
        className="bottom-left"
      >
        Anna Gunn
      </span>
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Skyler White
      </h5>
      <p
        className="card-text"
      >
        House wife, Book Keeper, Car Wash Manager, Taxi Dispatcher
      </p>
      <p
        className="card-text"
      >
        <small
          className="text-muted"
        >
          Appearance Seasons:
           
          1, 2, 3, 4, 5
        </small>
      </p>
    </div>
  </div>
</div>
`;
